xoxzo/cloudpy/cloudpy.py

```diff
@@ -199,6 +202,143 @@ def get_simple_playback_status(self, callid):
+            elif req_res.status_code == 404:
+                xr = XoxzoResponse(errors=req_res.status_code)
+            else:
+                xr = XoxzoResponse(
+                    errors=req_res.status_code,
+                    message=req_res.json())
+            return xr
+
+    def unsubscribe_din(self, din_uid):
+        """
+        subscribe DIN
+        :param din_uid:
+        :return:
+        """
+        url = self.xoxzo_api_dins_url + 'subscriptions/' + din_uid + '/'
+
+        try:
+            req_res = requests.delete(url, auth=(self.sid, self.auth_token))
+        except requests.exceptions.RequestException as e:
+            xr = XoxzoResponse(errors=XoxzoClient.REQUESTS_EXCEPITON, message={"http_error": e})
+            return xr
+        else:
+            if req_res.status_code == 200:
+                xr = XoxzoResponse()
+            elif req_res.status_code == 404:
+                xr = XoxzoResponse(errors=req_res.status_code)
+            else:
+                xr = XoxzoResponse(
+                    errors=req_res.status_code,
+                    message=req_res.json())
+            return xr
+
+    def get_subscription_list(self):
+        """
+        Get the list of the current subscribed DINs
+        :return:
+        """
+        url = self.xoxzo_api_dins_url + 'subscriptions/'
+
+        try:
+            req_res = requests.get(url, auth=(self.sid, self.auth_token))
+        except requests.exceptions.RequestException as e:
+            xr = XoxzoResponse(errors=XoxzoClient.REQUESTS_EXCEPITON, message={"http_error": e})
+            return xr
+        else:
+            if req_res.status_code == 200:
+                xr = XoxzoResponse(messages=req_res.json())
+            elif req_res.status_code == 404:
+                xr = XoxzoResponse(errors=req_res.status_code)
+            else:
+                xr = XoxzoResponse(
+                    errors=req_res.status_code,
+                    message=req_res.json())
+            return xr
+
+    def set_action_url(self, din_uid, action_url):
+        """
+        set action url to the din_uid
+        :param din_uid:
+        :param action_url:
+        :return:
+        """
+        url = self.xoxzo_api_dins_url + 'subscriptions/' + din_uid + '/'
+
+        payload = {
+            'action_url': action_url
+        }
+
+        try:
+            req_res = requests.post(url,
+                                    data=payload,
+                                    auth=(self.sid, self.auth_token))
+        except requests.exceptions.RequestException as e:
+            xr = XoxzoResponse(errors=XoxzoClient.REQUESTS_EXCEPITON, message={"http_error": e})
+            return xr
+        else:
+            if req_res.status_code == 200:
+                # set action url returns empty body when success
+                xr = XoxzoResponse()
+            elif req_res.status_code == 404:
+                xr = XoxzoResponse(errors=req_res.status_code)
```

Comment:
Started to see repetition in request handling logic. Maybe this can be refactor into single place. You can look this nexmo library as example - https://github.com/Nexmo/nexmo-python/blob/master/nexmo/__init__.py#L164